doc/source/general/cpu_hardware.rst

```diff
@@ -142,40 +146,134 @@ Triggers are all of the following:
-
-*As of kOS version 0.14 and higher, this condition is now being checked for* and the script will be **terminated with a runtime error** if the triggers like WHEN/THEN and ON take more than :attr:`Config:IPU` instructions to execute. The sum total of all the code within your WHEN/THEN and ON code blocks MUST be designed to complete within one physicd tick.
+For similar reasons to the explanation above about the ``WAIT`` command
+used inside triggers, it's not really a good idea for a trigger to
+have a long loop inside it that just keeps going and going.
 
-**This may seem harsh**. Ideally, kOS would only generate a runtime error if it thought your script was stuck in an **infinite loop**, and allow it to exceed the :attr:`Config:IPU` number of instructions if it was going to finish and just needed a little longer to to finish its work. But, because of a well known problem in computer science called `the halting problem <http://en.wikipedia.org/wiki/Halting_problem>`__, it's literally impossible for kOS, or any other software for that matter, to detect the difference between another program's infinite loop versus another program's loop that will end soon. kOS only knows how long your triggers have taken so far, not how long they're going to take before they're done, or even if they'll be done.
+The system does allow a trigger to take more than one **physics tick**
+to finish.  There are cases where it is entirely legitimate to do so
+if the trigger's body has too much work to do to get it all done in one
+update.  However, all triggers should be designed to finish their tasks
+in finite time and return.  What you should not do is design a trigger's
+body to go into an infinite loop, or a long-lasting loop that you thought
+would run in the background while the rest of the program continues on.
 
-If you suspect that your trigger body would have ended if it was allowed to run a little longer, try setting your :attr:`Config:IPU` setting a bit higher and see if that makes the error go away.
+This is because while you are in a trigger, ALL the other triggers aren't
+being fired, and the main-line code isn't being executed.  A trigger that
+performs a long-running loop will starve the rest of the code in your
+kerboscript program from being allowed to ever run again.
 
-If it does not make the error go away, then you will need to redesign your script to not depend on running a long-lasting amount of code inside triggers.
+This is a consequence of: *Main-line code gets interrupted by triggers,
+but triggers don't get interrupted by main-line code (or other triggers).*
```

Comment:
It might be good to highlight the fact that a long running loop in triggers will specifically disable proper steering control.  While the explanation makes it clear that steering controls are triggers, this may warrant a special note.